recipes/libelfin/all/CMakeLists.txt

```
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.1)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+project(${CONAN_PACKAGE_NAME} VERSION ${CONAN_PACKAGE_VERSION})
+conan_basic_setup()
```

```suggestion
cmake_minimum_required(VERSION 3.1)
project(cmake_wrapper)

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup()
```

Just to be sure that the include can depend on the detected compiler.
e.g. on Apple, the inclusion calls `conan_find_apple_framework`, which calls `find_library`, which is dependent on the current compiler.

---

Thank you for clarification. I would expect the `conanbuildinfo.cmake` to only set variables and have the `conan_*setup()` to run any heavy-weight routines. This way we could feed project name and version to `project()` and make better use of CMAKE_PROJECT_VERSION and CMAKE_PROJECT_NAME variables.